### DISCUSSION TITLE
"Create user" documentation update

### DISCUSSION THREAD

**Discussion by janzheng** [pocketbase/pocketbase#158] [Q&A]
This might stem from my misunderstanding of Pocketbase, but could we add something like "users need to log in before adding profile data" into the docs? I was struggling until I realized the user wasn't logged in.

Also, separately, could we have another class of "access restricted" errors? I think it's not always clear what access level you have when manipulating the database, e.g. updating records.

Something like this:
```import PocketBase from 'pocketbase';

const client = new PocketBase('http://localhost:8090');

...

// create user
const user = await client.Users.create({
    email: "test@example.com",
    password: "123456",
    passwordConfirm: "123456",
});

// authenticate & log in the user before setting profile data 
await client.Users.authViaEmail(user.email, user.password);

// set user profile data
const updatedRecord = await client.Records.update("profiles", user.profile.id, {
    name: "test",
});

// send verification email
await client.Users.sendVerificationEmail(user.email);
```

**Comment by ganigeorgiev** [comment]
We can add it, but that's not always true because the user profile is a regular collection with its own rules and developers are allowed to change this behavior.

Here - https://pocketbase.io/docs/manage-users/#custom-fields, below the screenshot, there is an example that shows a simple use case where this is useful.

You can view and change the user profile API rules from "Users -> Edit profile collection" cogwheel.

**Reply by janzheng** [reply]
Oh that's really interesting! Ok that makes a lot of sense now. I think I got tripped up thinking that no auth was needed, and got an error that wasn't related to "no access". Maybe something along these lines could prevent others from getting tripped up?

1. Add "unauthorized" error for actions prevented by the API rules 
2. Maybe adding a link to the Create User docs about default Users and Profile collection behavior, when updating profile?

I think I also got tripped up that Profiles are just a Collection type that doesn't appear on the dashboard, and are edited through the Users tab / Users collection. 

Thanks so much!
